### ISSUE TITLE
Bad callback logic

### ISSUE THREAD

**Issue by theuni** [bitcoin/secp256k1#300]
Given that ARG_CHECK is defined as:
```c
#define ARG_CHECK(cond) do { \
    if (EXPECT(!(cond), 0)) { \
        ctx->illegal_callback.fn(#cond, ctx->illegal_callback.data); \
        return 0; \
    } \
} while(0)
```
These checks (17 of 'em from a quick grep) make no sense and are doomed to explode:
```c
ARG_CHECK(ctx != NULL);
```

**Comment by gmaxwell** [comment]
Haha.

**Comment by sipa** [comment]
Good catch!

**Comment by sipa** [comment]
Fixed by #301
